src/routeBuild.js

```diff
@@ -31,7 +36,10 @@ function match(path, params, pathname) {
   for (let i = 1, len = m.length; i < len; ++i) {
     const key = keys[i - 1];
     const val = typeof m[i] === 'string' ? decodeURIComponent(m[i]) : m[i];
-    if (key) params[key.name] = val;
+    if (key) {
+      // TODO: fix mutation of input parameter
```

Comment:
Generally a very bad practice to change input object, if this function is ever to be reused somewhere else it should be refactored so `params` is returned instead